### PR TITLE
Read back pressure configs from config file

### LINKED ISSUE #1750: Backpressure configs are hard coded
The [documentation](http://twitter.github.io/heron/docs/operators/configuration/stmgr/) for the stream manage mentions the ability to configure specific different parameters related to the backpressure algorithm however it appears that the actual values are hard coded (see https://github.com/twitter/heron/blob/master/heron/common/src/cpp/network/connection.cpp#L27-L38).

It looks like the [methods](https://github.com/twitter/heron/blob/master/heron/common/src/cpp/network/connection.cpp#L27-L38) for accessing the config values exist but they are not being used. Heron should use the supplied values instead of having them hard coded.

### PR DESCRIPTION
**Pull request by congwang**
heron.streammgr.network.backpressure.highwatermark.mb and heron.streammgr.network.backpressure.lowwatermark.mb are already available in heron_internals.yaml, but we never actually read them from it. This PR makes stream manager reading these configs from heron_internals.yaml. Note, other non-stmgr connections still use the default values in the original code.

This resolves #1750.

### CODE REVIEW

File: heron/stmgr/src/cpp/server/stmgr-main.cpp [hunk 1]
```diff
@@ -70,8 +70,15 @@ int main(int argc, char* argv[]) {
     LOG(FATAL) << "Corrupt topology defn file" << std::endl;
   }
 
+  sp_int64 high_watermark = heron::config::HeronInternalsConfigReader::Instance()
+                              ->GetHeronStreammgrNetworkBackpressureHighwatermarkMb() *
+                                1024 * 1024;
```
Comment (kramasamy) [inline heron/stmgr/src/cpp/server/stmgr-main.cpp]: Instead of multiplying by 1024* 1024 - every time, can you define a constant in one of the include file called 

const int MB = 1024 * 1024;

somewhere in the include file

File: heron/stmgr/tests/cpp/server/stmgr_unittest.cpp [hunk 2]
```diff
@@ -1445,11 +1440,6 @@ TEST(StMgr, test_tmaster_restart_on_same_address) {
 
   int num_msgs_sent_by_spout_instance = 100 * 1000 * 1000;  // 100M
```
Comment (kramasamy) [inline heron/stmgr/tests/cpp/server/stmgr_unittest.cpp]: This is not 100 MB it should be 100 * 1024 * 1024

Comment (congwang) [inline heron/stmgr/tests/cpp/server/stmgr_unittest.cpp]: This is not my code, it shows up as diff context... Also, the 100M there might mean 100 million rather than 100 megabytes.

File: heron/stmgr/src/cpp/server/stmgr-main.cpp (hunk 1 above)
Comment (congwang) [inline heron/stmgr/src/cpp/server/stmgr-main.cpp]: Yeah, we can do it in a separate PR, because there are multiple places using the same.

File: heron/stmgr/tests/cpp/server/stmgr_unittest.cpp (hunk 2 above)
Comment (kramasamy) [inline heron/stmgr/tests/cpp/server/stmgr_unittest.cpp]: got it. my bad.

File: heron/stmgr/src/cpp/manager/stmgr-clientmgr.cpp [hunk 3]
```diff
@@ -114,6 +117,8 @@ StMgrClient* StMgrClientMgr::CreateClient(const sp_string& _other_stmgr_id,
   options.set_max_packet_size(config::HeronInternalsConfigReader::Instance()
                                   ->GetHeronStreammgrNetworkOptionsMaximumPacketMb() *
                               1024 * 1024);
+  options.set_high_watermark(high_watermark_);
```
Comment (maosongfu) [inline heron/stmgr/src/cpp/manager/stmgr-clientmgr.cpp]: Should we pass high_watermark and low_watermark from constructor, or similar to max_packet_size, reading config inside CreateClient(...) method?

On the one hand, it is not necessary to change the signature of a class's constructor. On the other hand, It will match the style of max_packet_size.

Comment (congwang) [inline heron/stmgr/src/cpp/manager/stmgr-clientmgr.cpp]: We have to, because the stmgr unit test cases don't want to read from config, they set these values by themselves.

Comment (maosongfu) [inline heron/stmgr/src/cpp/manager/stmgr-clientmgr.cpp]: Just wondering: how does the "max_packet_size" config work?

Comment (congwang) [inline heron/stmgr/src/cpp/manager/stmgr-clientmgr.cpp]: 1) StMgr() doesn't accept NetworkOptions as parameter, max_packet_size is passed in NetworkOptions for those who accept it.

2) stmgr unit tests want 2 different values in different tests, so you have to provide at least 2 different configs if you really want to read from config. Not to mention other config's.

Comment (maosongfu) [inline heron/stmgr/src/cpp/manager/stmgr-clientmgr.cpp]: OK Sounds good to me.

File: heron/stmgr/src/cpp/server/stmgr-main.cpp (hunk 1 above)
Comment (kramasamy) [inline heron/stmgr/src/cpp/server/stmgr-main.cpp]: can you please open an issue to track this?

Comment (congwang) [inline heron/stmgr/src/cpp/server/stmgr-main.cpp]: Done: https://github.com/twitter/heron/issues/1758
